docs/madr/decisions/011-mesh-traffic-route.md

```diff
@@ -0,0 +1,555 @@
+requests or traffic, modifies it in some way with _filters_ and then sends it to a _backend_.
+
+#### Matching
+
+The new route matches requests based on:
+
+- headers
+- path
+- method
+- query parameters
+
+##### Headers
+
+Headers can be tested for:
+
+- presence
+- absence
+- exact value
+- prefix value
+- regex value
+
+##### Spec
+
+A rule contains a list of match objects, each of which specifies a match for one
+or more of the above types:
+
+```yaml
+  matches:
+  - methods:
+    - GET
+    - POST
+    path:
+      type: RegularExpression|Exact|Prefix
+      value: ...
+    headers:
+      - type: RegularExpression|Exact|Prefix|Absent
+        name: "..."
+        value: ...
+    queryParams:
+      - type: RegularExpression|Exact
+        name: "..."
+        value: ...
+```
+
+A given entry in `match` succeeds if _all_ of the conditions are satisfied and the rule matches when _any one_ of the entries under `match` suceeds. In other words, `match` is an OR'd list of AND'd conditions.
+
+#### Request modification
+
+With `filters` we can change:
+
+- request headers
+- response headers
+- request mirroring (response is ignored)
+- HTTP redirect
+- host
+- request path
+
+```yaml
+  filters:
+  - type: RequestHeaderModifier|ResponseHeaderModifier|RequestMirror|RequestRedirect|URLRewrite
+    requestHeaderModifier:
+      set:
+      - name:
+        value:
+      add:
+      - name:
+        value:
+      remove:
+      - "name"
+    responseHeaderModifier:
+      set:
+      - name:
+        value:
+      add:
+      - name:
+        value:
+      remove:
+      - name
+    requestMirror:
+      backendRef:
+        kind: MeshService
+        name: svc_name
+    requestRedirect:
+      scheme:
+      hostname:
+      path:
+      port:
+      statusCode:
+    urlRewrite:
+      hostname:
+      path:
+        replaceFullPath:
+        replacePrefixMatch:
+```
+
+#### Routing
+
+If `matches` succeeds, the request is routed to the specified weighted destinations.
+
+The destinations appear as references in `backendRefs`.
+In the case that a request should be routed to the service it would otherwise
+have been routed to, `backendRefs` can be omitted.
+
+```yaml
+- backendRefs:
+  - weight: 90
+    kind: MeshServiceSubset
+    name: svc_name
+    tags:
+      version: v2
+  - weight: 10
+    kind: MeshServiceSubset
+    name: svc_name_other
+    tags:
+      version: v2
+    filters:
+      - ...
+  ...
+```
+
+Note in the above that a subset of filters (TODO which) are also available _per-backend_ basis.
+This also allows us to later support `HTTPBackendRef.filters` with Gateway API.
+
+#### Load balancing
+
+At the moment `TrafficRoute` can be used to configure load balancing behavior at
+the top, rule-independent level. This MADR proposes load balancing as a _filter_.
+This way it can be applied on a _per-route_ basis as well as a _per-backend_ basis.
+
+```yaml
+spec:
+  default:
+    rules:
```

Comment:
If loadbalancing is a `rule-independent level` shouldn't this be outside of rules?

Reply:
I removed it from the top level. I discussed this with @lahabana. We might also consider some way of referencing common configuration in the future a la logging backends.